CMakeLists.txt

```diff
@@ -551,6 +552,37 @@ if(NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/nnvm/CMakeLists.txt")
   list(APPEND mxnet_LINKER_LIBS ${nnvm_LINKER_LIBS})
 endif()
 
+# ---[ NNPack
+if(USE_NNPACK)
+  if (USE_MKLDNN)
+    message(WARNING "With MKLDNN enabled NNPack operators will not run.")
```

Comment:
Why? What happens in that case?

Reply:
both MKLDNN and NNPack define CPU versions of some operators, while there won't be an issue with compilation, when these operators are invoked they will need to use only the one backend.

Reply:
What happens if a user invokes one of these operators?

Reply:
The way I'm coding it, it will use the mkldnn backend. Once I finish that maybe some testing would show that NNPack is faster in which case it can be changed to the default.

Reply:
I'm not a fan of such implicit actions. It should be up to the user to decide which Backend to use. To me, setting both options seems like a conflict and should fail.

Reply:
I'll change it to fail in that case.